app.py

```diff
@@ -202,19 +202,29 @@ def index(category=[]):
     groups = []
     if category:
         groups = _get_groups_by_slug(slugs=category)
+        if not groups:
+            groups.append({
+                "id": -1,
+                "name": category
+            })
 
     page = flask.request.args.get('page')
     posts, metadata = _get_posts(
         groups=[str(group['id']) for group in groups],
         page=page
     )
     if category:
-        return flask.render_template(
-            'group.html',
-            posts=posts,
-            category=groups[0] if groups else None,
-            **metadata
-        )
+        if posts:
```

Comment:
Shouldn't you check that the group's id is `-1` instead of checking for the existence of posts. There could be a group with no posts.

Reply:
I moved the redirection upper in the code so that when a group doesn't exists, the sever doesn't try to catch any post.